client/providers/public.go

```diff
@@ -0,0 +1,30 @@
+package providers
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+
+	"github.com/openpubkey/openpubkey/client"
+)
+
+type PublicKeyResolver func(context.Context, []byte) (crypto.PublicKey, error)
+
+type publicKeyResolver struct {
+	client.OpenIdProvider
+	resolver PublicKeyResolver
+}
```

Comment:
I'm not sure this adds much - I think it's reasonable for the calling code to define their own struct which embeds `client.OpenIdProvider` and add their own implementation of `PublicKey`. WDYT?